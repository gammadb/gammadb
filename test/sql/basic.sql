--
-- Regression Tests for Custom Plan APIs
--

create extension gammadb;
-- construction of test data

CREATE TABLE t1 (
    a   int,
	b	double precision
) using gamma;
INSERT INTO t1 SELECT generate_series(1,3), 2.3;
INSERT INTO t1 SELECT generate_series(1,3), 3.3;
INSERT INTO t1 SELECT generate_series(1,3), 4.3;
VACUUM ANALYZE t1;

set work_mem = '100MB';

EXPLAIN (COSTS OFF) SELECT * FROM t1;
SELECT * FROM t1;

EXPLAIN (COSTS OFF) SELECT b FROM t1;
SELECT b FROM t1;

EXPLAIN (COSTS OFF) SELECT b+1 FROM t1;
SELECT b+1 FROM t1;

EXPLAIN (COSTS OFF) SELECT count(b) FROM t1;
SELECT count(b) FROM t1;

set enable_hashagg to on;

EXPLAIN (COSTS OFF) SELECT a, sum(b), avg(b)  FROM t1 group by a;
SELECT a, sum(b), avg(b)  FROM t1 group by a;

EXPLAIN (COSTS OFF) SELECT a, sum(b), avg(b)  FROM t1 where a < 3 group by a;
SELECT a, sum(b), avg(b)  FROM t1 where a < 3 group by a;

set enable_hashagg to off;

EXPLAIN (COSTS OFF) SELECT a, sum(b), avg(b)  FROM t1 group by a;
SELECT a, sum(b), avg(b)  FROM t1 group by a;

EXPLAIN (COSTS OFF) SELECT a, sum(b), avg(b)  FROM t1 where a < 3 group by a;
SELECT a, sum(b), avg(b)  FROM t1 where a < 3 group by a;

reset enable_hashagg;

drop table t1;

drop extension gammadb;
