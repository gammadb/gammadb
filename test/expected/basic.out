--
-- Regression Tests for Custom Plan APIs
--
create extension gammadb;
-- construction of test data
CREATE TABLE t1 (
    a   int,
	b	double precision
) using gamma;
INSERT INTO t1 SELECT generate_series(1,3), 2.3;
INSERT INTO t1 SELECT generate_series(1,3), 3.3;
INSERT INTO t1 SELECT generate_series(1,3), 4.3;
VACUUM ANALYZE t1;
set work_mem = '100MB';
EXPLAIN (COSTS OFF) SELECT * FROM t1;
               QUERY PLAN                
-----------------------------------------
 Custom Scan (gamma_vec_devector)
   ->  Custom Scan (gamma_vec_tablescan)
         ->  Seq Scan on t1
(3 rows)

SELECT * FROM t1;
 a |  b  
---+-----
 1 | 2.3
 2 | 2.3
 3 | 2.3
 1 | 3.3
 2 | 3.3
 3 | 3.3
 1 | 4.3
 2 | 4.3
 3 | 4.3
(9 rows)

EXPLAIN (COSTS OFF) SELECT b FROM t1;
               QUERY PLAN                
-----------------------------------------
 Custom Scan (gamma_vec_devector)
   ->  Custom Scan (gamma_vec_tablescan)
         ->  Seq Scan on t1
(3 rows)

SELECT b FROM t1;
  b  
-----
 2.3
 2.3
 2.3
 3.3
 3.3
 3.3
 4.3
 4.3
 4.3
(9 rows)

EXPLAIN (COSTS OFF) SELECT b+1 FROM t1;
                  QUERY PLAN                   
-----------------------------------------------
 Result
   ->  Custom Scan (gamma_vec_devector)
         ->  Custom Scan (gamma_vec_tablescan)
               ->  Seq Scan on t1
(4 rows)

SELECT b+1 FROM t1;
 ?column? 
----------
      3.3
      3.3
      3.3
      4.3
      4.3
      4.3
      5.3
      5.3
      5.3
(9 rows)

EXPLAIN (COSTS OFF) SELECT count(b) FROM t1;
                  QUERY PLAN                   
-----------------------------------------------
 Custom Scan (gamma_vec_agg)
   ->  Aggregate
         ->  Custom Scan (gamma_vec_tablescan)
               ->  Seq Scan on t1
(4 rows)

SELECT count(b) FROM t1;
 count 
-------
     9
(1 row)

set enable_hashagg to on;
EXPLAIN (COSTS OFF) SELECT a, sum(b), avg(b)  FROM t1 group by a;
                  QUERY PLAN                   
-----------------------------------------------
 Custom Scan (gamma_vec_agg)
   ->  HashAggregate
         Group Key: a
         ->  Custom Scan (gamma_vec_tablescan)
               ->  Seq Scan on t1
(5 rows)

SELECT a, sum(b), avg(b)  FROM t1 group by a;
 a |        sum        |        avg         
---+-------------------+--------------------
 1 | 9.899999999999999 | 3.2999999999999994
 3 | 9.899999999999999 | 3.2999999999999994
 2 | 9.899999999999999 | 3.2999999999999994
(3 rows)

EXPLAIN (COSTS OFF) SELECT a, sum(b), avg(b)  FROM t1 where a < 3 group by a;
                        QUERY PLAN                        
----------------------------------------------------------
 Custom Scan (gamma_vec_agg)
   ->  HashAggregate
         Group Key: a
         ->  Custom Scan (gamma_vec_tablescan)
               ->  Seq Scan on t1
                     Filter: (a OPERATOR(pg_catalog.<) 3)
(6 rows)

SELECT a, sum(b), avg(b)  FROM t1 where a < 3 group by a;
 a |        sum        |        avg         
---+-------------------+--------------------
 1 | 9.899999999999999 | 3.2999999999999994
 2 | 9.899999999999999 | 3.2999999999999994
(2 rows)

set enable_hashagg to off;
EXPLAIN (COSTS OFF) SELECT a, sum(b), avg(b)  FROM t1 group by a;
                        QUERY PLAN                         
-----------------------------------------------------------
 Custom Scan (gamma_vec_agg)
   ->  GroupAggregate
         Group Key: a
         ->  Custom Scan (gamma_vec_sort)
               ->  Sort
                     Sort Key: a USING <
                     ->  Custom Scan (gamma_vec_tablescan)
                           ->  Seq Scan on t1
(8 rows)

SELECT a, sum(b), avg(b)  FROM t1 group by a;
 a |        sum        |        avg         
---+-------------------+--------------------
 1 | 9.899999999999999 | 3.2999999999999994
 2 | 9.899999999999999 | 3.2999999999999994
 3 | 9.899999999999999 | 3.2999999999999994
(3 rows)

EXPLAIN (COSTS OFF) SELECT a, sum(b), avg(b)  FROM t1 where a < 3 group by a;
                              QUERY PLAN                              
----------------------------------------------------------------------
 Custom Scan (gamma_vec_agg)
   ->  GroupAggregate
         Group Key: a
         ->  Custom Scan (gamma_vec_sort)
               ->  Sort
                     Sort Key: a USING <
                     ->  Custom Scan (gamma_vec_tablescan)
                           ->  Seq Scan on t1
                                 Filter: (a OPERATOR(pg_catalog.<) 3)
(9 rows)

SELECT a, sum(b), avg(b)  FROM t1 where a < 3 group by a;
 a |        sum        |        avg         
---+-------------------+--------------------
 1 | 9.899999999999999 | 3.2999999999999994
 2 | 9.899999999999999 | 3.2999999999999994
(2 rows)

reset enable_hashagg;
drop table t1;
drop extension gammadb;
